controller/api/public/client_test.go

```diff
@@ -31,7 +30,7 @@ func TestNewInternalClient(t *testing.T) {
 		mockTransport := &mockTransport{}
 		mockTransport.responseToReturn = &http.Response{
 			StatusCode: 500,
-			Body:       ioutil.NopCloser(strings.NewReader("body")),
+			Body:       ioutil.NopCloser(bufferedReader(t, &pb.Empty{})),
```

Comment:
2GB allocation in `deserializePayloadFromReader()`!
```golang
int(binary.LittleEndian.Uint32([]byte("body"))) == 2036625250
```

https://play.golang.org/p/5AXS4mFaKRW